packages/dynamic_layouts/lib/src/staggered_layout.dart

```diff
@@ -0,0 +1,279 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/rendering.dart';
+
+import '../dynamic_layouts.dart';
+
+/// A [DynamicSliverGridLayout] that creates tiles with varying main axis
+/// sizes and fixed cross axis sizes, generating a staggered layout. Similar to
+/// Android's StaggeredGridLayoutManager
+/// {https://developer.android.com/reference/androidx/recyclerview/widget/StaggeredGridLayoutManager}.
+///
+/// The tiles are placed in the column (or row in case [scrollDirection] is set
+/// to [Axis.horizontal]) with the minimum extent, which means children are not
+/// neccesarily layed out in sequential order.
+///
+/// See also:
+///  * [DynamicSliverGridDelegateWithMaxCrossAxisExtent] and
+///  * [DynamicSliverGridDelegateWithFixedCrossAxisCount], which use this
+///    layout.
+///  * [SliverGridLayout], which represents an arbitrary tile layout.
```

Comment:
This could probably be removed. It is less relevant to this package, and less likely to be used by the user as an abstract base class.